cpp/arcticdb/version/version_core.cpp

```diff
@@ -1172,6 +1209,7 @@ VersionedItem sort_merge_impl(
     auto num_versioned_rows = pipeline_context->total_rows_;
 
     read_incompletes_to_pipeline(store, pipeline_context, read_query, ReadOptions{}, convert_int_to_float, via_iteration, sparsify);
+    check_incompletes_index_ranges_dont_overlap(pipeline_context);
```

Comment:
The docs for the top-level `sort_merge` API state: "sort_merge will sort and merge incomplete segments. The segments do not have to be ordered - incomplete segments can contain interleaved time periods but the final result will be fully ordered".

Does it make sense then to check they don't overlap therefore here if the method doesn't require that they are non-overlapping?